mistakedetection.tests/src/ca/mcgill/sel/mistakedetection/tests/MistakeDetectionPerformanceAnalysis.java

```diff
@@ -264,6 +342,298 @@ public void testStudentSolution20() {
+      int count = 1;
+      var instElem = getConcatNames(m.getInstructorElementNames());    
+      var studElem = getConcatNames(m.getStudentElementNames());
+
+      var mistakeType = m.getMistakeType().getName();
+      for (Mistake m1 : notExtraMistakes) {
+        if (m != m1) {
+          var instElem1 = getConcatNames(m1.getInstructorElementNames());          
+          var studElem1 = getConcatNames(m1.getStudentElementNames());
+         
+          if (instElem.equals(instElem1) && studElem.equals(studElem1)
+              && !(m.getMistakeType().getName().equals(WROMG_MULTIPLICTY)
+                  && m1.getMistakeType().getName().equals(WROMG_ROLE_NAME))
+              && !(m.getMistakeType().getName().equals(WROMG_ROLE_NAME)
+                  && m1.getMistakeType().getName().equals(WROMG_MULTIPLICTY))) {
+            mistakeType += ", " + m1.getMistakeType().getName();
+            count++;
+          }
+        }
+      }
+      var fnlString = instElem + studElem;
+      if (!rendered.contains(fnlString) || mistakeType.equals(WROMG_ROLE_NAME) || mistakeType.equals(WROMG_MULTIPLICTY)) {
+        id++;
+        studentData.put(String.valueOf(id),
+            new Object[] {instElem, studElem, mistakeType, "", count, "=D" + id + "=E" + id, ""});
+        rendered.add(fnlString);
+      }
+    }
+    int instructorElems = id - 1;
+    int extraElemsStart = id;
+    rendered = new ArrayList<>();
+    for (Mistake m : extraMistakes) {
+      int count = 1;
+      var instElem = getConcatNames(m.getInstructorElementNames());    
+      var studElem = getConcatNames(m.getStudentElementNames());
+      
+      var mistakeType = m.getMistakeType().getName();
+      for (Mistake m1 : extraMistakes) {
+        if (m != m1) {
+          var instElem1 = getConcatNames(m1.getInstructorElementNames());    
+          var studElem1 = getConcatNames(m1.getStudentElementNames());
+          if (instElem.equals(instElem1) && studElem.equals(studElem1)
+              && !(m.getMistakeType().getName().equals(WROMG_MULTIPLICTY)
+                  && m1.getMistakeType().getName().equals(WROMG_ROLE_NAME))
+              && !(m.getMistakeType().getName().equals(WROMG_ROLE_NAME)
+                  && m1.getMistakeType().getName().equals(WROMG_MULTIPLICTY))) {
+            mistakeType += ", " + m1.getMistakeType().getName();
+            count++;
+          }
+        }
+      }
+      var fnlString = instElem + studElem;
+      if (!rendered.contains(fnlString) || mistakeType.equals(WROMG_ROLE_NAME) || mistakeType.equals(WROMG_MULTIPLICTY)) {
+        studentData.put(String.valueOf(id),
+            new Object[] {instElem, studElem, mistakeType, "", count, "=D" + id + "=E" + id, ""});
+        id++;
+        rendered.add(fnlString);
+      }
+    }
+
+    int nId = id;
+    studentData.put(String.valueOf(nId++), new Object[] {"", "", "", "", "", "", ""});
+    studentData.put(String.valueOf(nId++), new Object[] {"", "", "", "", "", "", ""});
```

Comment:
Is there a better way to create this object array? In Python we can do `7 * [""]` for this